### ISSUE TITLE
windowlist@cobinja.de: Can't pin apps

### ISSUE THREAD

**Issue by vertigo220** [linuxmint/cinnamon-spices-applets#1654]
```
 * Applet version/Build date (to be found in "About.." or in "metadata.json") - 2018/01/08
 * Cinnamon version (cinnamon --version) - 3.6.7
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) - Mint 18.3
 * Graphics hardware *and* driver used - i3 integrated / unknown
 * Snippet or attachment of ~/.cinnamon/glass.log

[glass.log](https://github.com/linuxmint/cinnamon-spices-applets/files/1786485/glass.log)

 * Snippet or attachment of ~/.xsession-errors

[xsession-errors.txt](https://github.com/linuxmint/cinnamon-spices-applets/files/1786489/xsession-errors.txt)


 ```

**Notify author of applet**
@Cobinja 

**Issue**

This applet clearly has a "pinning" feature (favorites), but when I added it to the panel, it only had two apps pinned. I'm assuming it pins whatever apps are in the favorites in the menu, but then it doesn't make sense only two were pinned as I have several favorite app there. Also, it would be nice to be able to pin any apps via their button's context menu.

**Steps to reproduce**

N/A

**Expected behaviour**

To be able to select pinning options via context menu for all buttons on panel.

**Other information**

**Comment by vertigo220** [comment]
Update: apparently, for some apps the option to pin them becomes available right away, whereas others (e.g. Pale Moon, Notepadqq, and likely others) it's unavailable until the app is closed and reopened.

**Comment by Cobinja** [comment]
The pinning feature depends on whether the application has a .desktop file

**Comment by vertigo220** [comment]
I don't think that explains (or maybe it does, this is all still pretty new to me) why the feature suddenly became available after closing and reopening the apps. Does CWL add the file for all apps, but it doesn't take effect until the app is restarted? Perhaps there should be a note in CWL's settings that some apps may not have the option until they're restarted?

**Comment by Cobinja** [comment]
This applet does not add any desktop files, it's just not the right place to do that.
You can add desktop files via cinnamon's main menu's settings -> "Menu" -> "Open the menu editor"

**Comment by vertigo220** [comment]
In that case I don't know why some of the apps only showed the pin option after restarting them. I'm not too worried about it since it seems like it _does_ in fact work, but there appears to be something "off" in the process.
